### PR TITLE
make Node.toString stack safe

### PR DESCRIPTION
**Pull request by lrytz**

### CODE REVIEW

File: shared/src/main/scala/scala/xml/Utility.scala [hunk 1]
```diff
@@ -212,32 +212,64 @@ object Utility extends AnyRef with parsing.TokenTests {
...
+        if (toClose.head != null) {
           sb.append("</")
-          el.nameToString(sb)
+          toClose.head.nameToString(sb)
           sb.append('>')
         }
-      case _ => throw new IllegalArgumentException("Don't know how to serialize a " + x.getClass.getName)
+        ser(rests, pscopes.tail, spaced.tail, toClose.tail)
+      case ns1 :: r =>
+        val (n, ns) = (ns1.head, ns1.tail)
```
[the remark this answers is not in the document]
Comment (dubinsky) [reply shared/src/main/scala/scala/xml/Utility.scala:237]: Up to you; I am not too concerned about the performance of `toString` methods.

**Review by dubinsky — APPROVED** [review]
Wow! LGTM.

File: shared/src/main/scala/scala/xml/Utility.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (lrytz) [reply shared/src/main/scala/scala/xml/Utility.scala:237]: I'll add it.

**Comment by dubinsky** [comment]
> @dubinsky feel free to go ahead. I might backport it to 1.x. Do you have any releases (1.x / 2.x) planned?

I have one more change planned before the 2.2 release.
No 1.x releases are planned that I know of - @SethTisue?

**Pull request merged**